sympy/logic/boolalg.py

```diff
@@ -2033,6 +2033,11 @@ def simplify_logic(expr, form=None, deep=True):
     truthtable = []
```

Comment:
What about finding where the non-True/False values are once?

```
ix=[i for i in range(len(variables)) if i not in (0,1)]
for T in product([0, 1], repeat=len(ix)):
    t = list(variables)
    for j,i in enumerate(ix):
        t[i] = T[j]
...
```

Reply:
what i get from your code is -
ix=[2, 3, 4, ....] will be something like that , so this is for only two fixed  position[0,1] "True/False" case.
and other problem is ,"variables" is a set so position of "True/False" is not fixed so if we make list of variable it may be different for the same set e.g variables={a, False, True, b, c} and other time it comes { False, True,a, b, c}  for the same expression so if we make list it will be different .
so we can't find position of "True/False" directly.
that's why i am doing in naive way if you have other solution please tell me it will be very helpful.

Reply:
How about this?
```python
    variables = list(ordered(_find_predicates(expr)))  # atoms to the left
    truthtable = []
    i = 0
    while variables[i] in (True, False):  # detect 1, 0, True, False, S.true, S.false
        variables[i] = 1 if variables[i] == True else 0
        i += 1
    for t in product([0, 1], repeat=len(variables) - i):
        if expr.xreplace(dict(zip(variables[i:], t))) == True:
            truthtable.append(variables[:i] + t[:])
```

Reply:
i have done some editing based on your code ,this time it seems better to me.

Reply:
@smichr I do according to your code this time it seems good to me.